main.py

```diff
@@ -1,129 +1,156 @@
+from datetime import datetime, timedelta
 import requests
 import os
 from dotenv import load_dotenv
 from flask import Flask, jsonify
-from datetime import datetime
 from sql_client import SQLClient
 
 load_dotenv()
 
+api_key = os.getenv('WEATHER_API_KEY')
+api_url_base = os.getenv('WEATHER_API_URL', 'https://dataservice.accuweather.com')
+
 ipinfo_api_token = os.getenv('IPINFOAPITOKEN')
 
 app = Flask(__name__)
 
 db_client = SQLClient()
 
+# Simple cache to store weather data
+weather_cache = {}
+
+
 def create_weather_table():
     create_table_query = """
     CREATE TABLE IF NOT EXISTS weather (
         id INT AUTO_INCREMENT PRIMARY KEY,
-        bikeable BOOLEAN,
         icon VARCHAR(255),
         lat FLOAT,
         lng FLOAT,
         saved_at DATETIME,
         temperature DECIMAL(5, 2)
     );
     """
-    db_client.query_fix(create_table_query)
+    try:
+        db_client.query_fix(create_table_query)
+    except Exception as e:
+        print(f"Table was not able to set error {e}")
+
+
+def insert_weather_data(weather_data: dict):
+    try:
+        db_client.insert(
+            keys=('icon', 'lat', 'lng', 'saved_at', 'temperature'),
+            values=(
+                weather_data['icon'],
+                weather_data['location']['lat'],
+                weather_data['location']['lng'],
+                weather_data['saved_at'],
+                weather_data['temperature']
+            ),
+            table='weather'
+        )
+    except Exception as e:
+        print(f"Something went wrong with inputting the value into the database \n Error: {e}")
```

Comment:
If a type error is thrown or a insert error is not returning the error to the user. 9 out of 10 occasions you would not return the error to the user

Reply:
fixed